### ISSUE TITLE
Performance 2022

### ISSUE THREAD

**Issue by rviscomi** [HTTPArchive/almanac.httparchive.org#2890]
# Performance 2022

![Performance illustration](https://raw.githubusercontent.com/HTTPArchive/almanac.httparchive.org/main/src/static/images/2019/performance/hero_lg.jpg)

If you're interested in contributing to the Performance chapter of the 2022 Web Almanac, please reply to this issue and indicate which role or roles best fit your interest and availability: [author](https://github.com/HTTPArchive/almanac.httparchive.org/wiki/Authors'-Guide), [reviewer](https://github.com/HTTPArchive/almanac.httparchive.org/wiki/Reviewers'-Guide), [analyst](https://github.com/HTTPArchive/almanac.httparchive.org/wiki/Analysts'-Guide), and/or [editor](https://github.com/HTTPArchive/almanac.httparchive.org/wiki/Editors'-Guide).

## Content team

| Lead | Authors | Reviewers | Analysts | Editors | Coordinator |
| ---- | ------- | --------- | -------- | ------- | ----------- |
| @mel-ada | @mel-ada | @konfirmed @rviscomi @25prathamesh @estelle | @konfirmed @25prathamesh @siakaramalegos  | - | @foxdavidj |

<details>
<summary><b>Expand for more information about each role 👀</b></summary>

- The **[content team lead](https://github.com/HTTPArchive/almanac.httparchive.org/wiki/Content-Team-Leads'-Guide)** is the chapter owner and responsible for setting the scope of the chapter and managing contributors' day-to-day progress.
- **[Authors](https://github.com/HTTPArchive/almanac.httparchive.org/wiki/Authors'-Guide)** are subject matter experts and lead the content direction for each chapter. Chapters typically have one or two authors. Authors are responsible for planning the outline of the chapter, analyzing stats and trends, and writing the annual report.
- **[Reviewers](https://github.com/HTTPArchive/almanac.httparchive.org/wiki/Reviewers'-Guide)** are also subject matter experts and assist authors with technical reviews during the planning, analyzing, and writing phases.
- **[Analysts](https://github.com/HTTPArchive/almanac.httparchive.org/wiki/Analysts'-Guide)** are responsible for researching the stats and trends used throughout the Almanac. Analysts work closely with authors and reviewers during the planning phase to give direction on the types of stats that are possible from the dataset, and during the analyzing/writing phases to ensure that the stats are used correctly.
- **[Editors](https://github.com/HTTPArchive/almanac.httparchive.org/wiki/Editors'-Guide)** are technical writers who have a penchant for both technical and non-technical content correctness. Editors have a mastery of the English language and work closely with authors to help wordsmith content and ensure that everything fits together as a cohesive unit.
- The **[section coordinator](https://github.com/HTTPArchive/almanac.httparchive.org/wiki/Section-Leads'-Guide)** is the overall owner for all chapters within a section like "User Experience" or "Page Content" and helps to keep each chapter on schedule.

_Note: The time commitment for each role varies by the chapter's scope and complexity as well as the number of contributors._

For an overview of how the roles work together at each phase of the project, see the [Chapter Lifecycle](https://github.com/HTTPArchive/almanac.httparchive.org/wiki/Chapter-Lifecycle) doc.
</details>

### [**0. Form the content team**](https://github.com/HTTPArchive/almanac.httparchive.org/wiki/Chapter-Lifecycle#0-create-content-team)
+ [x] **May 1**: The content team has at least one author, reviewer, and analyst

### [**1. Plan content**](https://github.com/HTTPArchive/almanac.httparchive.org/wiki/Chapter-Lifecycle#1-plan-content)
+ [x] **May 15** The content team has completed the chapter outline in the [draft doc][~google-doc]

### [**2. Gather data**](https://github.com/HTTPArchive/almanac.httparchive.org/wiki/Chapter-Lifecycle#2-gather-data)
+ [x] **June 1**: Analysts have added all necessary [custom metrics](https://github.com/HTTPArchive/custom-metrics/blob/main/README.md) and drafted a PR ([example](https://github.com/HTTPArchive/almanac.httparchive.org/pull/1087)) to track query progress
- **June 1 - 15**: HTTP Archive runs the June crawl

### [**3. Validate results**](https://github.com/HTTPArchive/almanac.httparchive.org/wiki/Chapter-Lifecycle#3-validate-results)
+ [x] **August 1**: Analysts have queried all metrics and saved the output to the [results sheet][~google-sheets]

### [**4. Draft content**](https://github.com/HTTPArchive/almanac.httparchive.org/wiki/Chapter-Lifecycle#4-draft-content)
+ [ ] **September 1**: The content team has written, reviewed, and edited the chapter in the [doc][~google-doc]

### [**5. Publication**](https://github.com/HTTPArchive/almanac.httparchive.org/wiki/Chapter-Lifecycle#5-publication)
+ [ ] **September 15**: The completed chapter and all required [metadata](https://github.com/HTTPArchive/almanac.httparchive.org/wiki/Authors'-Guide#metadata-to-add-at-the-top-of-your-chapters) and [figures](https://github.com/HTTPArchive/almanac.httparchive.org/wiki/Figures-Guide) are converted to markdown and submitted to [GitHub][~chapter-markdown]
- **September 26**: Target launch date 🚀

## Chapter resources

Refer to these 2022 Performance resources throughout the content creation process:

[📄 Google Docs][~google-doc] for outlining and drafting content
[🔍 SQL files][~sql-files] for committing the queries used during analysis
[📊 Google Sheets][~google-sheets] for saving the results of queries
[📝 Markdown file][~chapter-markdown] for publishing content and managing public metadata
[💬 #web-almanac-performance on Slack][~slack-channel] for team coordination

[~google-doc]: https://docs.google.com/document/d/1IKV40fllCZTqeu-R6-73ckjQR9S6jiBfVBBfdcpAMkI/edit?usp=sharing
[~sql-files]: https://github.com/HTTPArchive/almanac.httparchive.org/tree/main/sql/2022/performance
[~google-sheets]: https://docs.google.com/spreadsheets/d/1TPA_4xRTBB2fQZaBPZHVFvD0ikrR-4sNkfJfUEpjibs/edit?usp=sharing
[~chapter-markdown]: https://github.com/HTTPArchive/almanac.httparchive.org/tree/main/src/content/en/2022/performance.md
[~slack-channel]: https://join.slack.com/t/httparchive/shared_invite/zt-45sgwmnb-eDEatOhqssqNAKxxOSLAaA

**Comment by konfirmed** [comment]
Hi, I'll like to be a Reviewer/Analyst in this chapter.

**Comment by morelme** [comment]
Hello, I'd like to be a Reviewer.

**Comment by morelme** [comment]
Hi, I'd like to be a co-author or reviewer.

**Comment by 25prathamesh** [comment]
HI, I am working in the web perf field for the past few years I like to be a co-author/reviewer/analyst.

**Comment by silent1mezzo** [comment]
Happy to help review/analyze data 👋

**Comment by foxdavidj** [comment]
Hey @morelme, would you be interested in taking the Chapter Lead role for the Performance chapter?

As the Chapter lead you'd be the primary author and the key person responsible for pulling the entire chapter together. Details on [the role and commitment here](https://github.com/HTTPArchive/almanac.httparchive.org/wiki/Content-Team-Leads'-Guide)

We'd love to have you 🎉 🎉

**Comment by morelme** [comment]
Hey @foxdavidj, that sounds great - excited to be part of this!

**Comment by mel-ada** [comment]
Before we get too far into this process, I figured I would update my handle to reflect a recent name change (last name from Morel -> Ada after getting married). It sounds like GitHub doesn't update tags, so if you click on my old handle and get a 404, don't worry. I'm still here! @morelme -> @mel-ada :)

**Comment by rviscomi** [comment]
@mel-ada @konfirmed @25prathamesh @silent1mezzo could you all request edit access to the [shared doc](https://docs.google.com/document/d/1IKV40fllCZTqeu-R6-73ckjQR9S6jiBfVBBfdcpAMkI/edit?usp=sharing)? I've added a few ideas for topics to cover but it'd be good to start iterating on it more collaboratively. And FYI we also have a [`#web-almanac-performance`](https://join.slack.com/t/httparchive/shared_invite/zt-45sgwmnb-eDEatOhqssqNAKxxOSLAaA) channel on Slack for higher bandwidth discussions.

**Comment by mel-ada** [comment]
@konfirmed @rviscomi @25prathamesh @silent1mezzo @foxdavidj 

Sounds great, it looks like almost everyone is on Slack so let's talk schedules there. cc @silent1mezzo, who I'm not sure has had the chance to join Slack yet!

**Comment by rviscomi** [comment]
Welcome back @estelle, great to have you on board as a reviewer!

**Comment by foxdavidj** [comment]
Hey @mel-ada, how are you feeling about completing the chapter outline by May 15th?

**Comment by mel-ada** [comment]
@foxdavidj Feeling good! During the kickoff, I think we had a good brainstorm that gave us a good pool of ideas to choose from. I'm hoping to meet as a team once more to discuss next week, if not we will finalize async.

**Comment by foxdavidj** [comment]
@mel-ada How are you feeling about the chapter outline? We're fast approaching the date where any new [custom metrics](https://github.com/HTTPArchive/custom-metrics/blob/main/README.md) need to be written, tested, and merged into the web crawler (May 27).

**Comment by foxdavidj** [comment]
@mel-ada Looks like the outline is coming along nicely. Can you specifically mark/notate what metrics may require new custom metrics together with your analyst? The deadline for implementing and getting them approved is in a week and a half.

**Comment by mel-ada** [comment]
@foxdavidj apologies, just seeing these comments - I will know to keep a closer eye on comms via GitHub from now on.

**Weekly Update**

✅  Outline:
The outline is in a good place - I would call this done.

🚧  Metrics List: 
I'm going through the document this weekend to compile the list of metrics needed, identify which ones need to be added, and delegate the implementation and testing to team members. I have started a section in the document for the metrics list. I will lean on team mates with past HTTP Archive experience to verify the feasibility of any custom metrics. I understand that this is high priority and the deadline (5/27) is coming up.

No questions at the moment - but may reach out with specific metric questions if anything comes up!

**Comment by mel-ada** [comment]
@foxdavidj 
**Weekly Update**

✅ Metrics List: 
This is complete in the planning doc. We worked together async to narrow down our list of metrics and identify custom ones. Then, I split the custom metrics up and assigned specific ones to our analysts for implementation. Rick even helped with a few (kudos)!

🚧 Custom Metrics:
We are working on hitting the hard deadline of 5/31 for custom metrics. You can follow along in the repository [here](https://github.com/HTTPArchive/custom-metrics/issues/16). Prathamesh has several metrics in flight that will be added to a branch shortly, and I'll be jumping in to help with the remaining.

**Comment by foxdavidj** [comment]
Can i just say how much i love the emojis? I need to start using them for things like this myself

**Comment by mel-ada** [comment]
**Weekly Update**
🚧 Non-Metric-Dependent Writing:
Still cooking - didn't have as much time this week as I would have liked.

🚧 Queries:
I checked in with @konfirmed, who said he'd be able to take the lead on this. Keeping an eye out for a draft PR and any questions that may come up.

**Comment by mel-ada** [comment]
Hello, what a time to be alive in the country that I live in! Here is an update: 

🐢 **Non-Metric-Dependent Writing:**
CWV Timeline is coming along. I grabbed an infographic template from Canva and added HTTP archive's branding colors (via color picker from the site). I'll pause on adding in content until I have that finalized, but it's linked in the planning doc incase anyone wants to take a look.
I'm also adding some commentary that is very unfinished and unstructured at the moment but I feel will more or less evolve into the intro.

🚧 **Queries:**
I don't see updates on the PR, so I'll turn to @konfirmed for updates here. Since all metrics are due to be queried in less than a month, how are you planning on tackling this? Have you talked with @25prathamesh to divide up the work? Let me know if you need any help! Reminder that there are some notes in the planning doc for what queries for each metric might look like.

**Comment by foxdavidj** [comment]
@mel-ada Could you link to the infographic you mentioned? Can't find it in the Google Doc. I might just be missing it though

**Comment by mel-ada** [comment]
@foxdavidj Here's [the link]( https://www.canva.com/design/DAFFe3DSZ7U/S9fC-_j3N6klv4_Jc3i2Ag/edit?utm_content=DAFFe3DSZ7U&utm_campaign=designshare&utm_medium=link2&utm_source=sharebutton). Disclaimer I am not a designer, would appreciate any design support if that's available to us! But doing what I can if not.

In the doc, it's easy to miss - this is a bullet point under 'Past: Core Web Vitals Rollout Timeline'.

**Comment by foxdavidj** [comment]
Looks quite nice. @mel-ada Sounds like your plan to use that image in the chapter? To give a summary of the timeline?

**Comment by mel-ada** [comment]
@foxdavidj Yes, one of the things I didn't find in my research is a clear timeline of all things CWV release and I think it will be helpful for folks to visualize. I'm waiting until the timeline content is finalized to switch out placeholder text in the image with the real deal.

**Comment by tunetheweb** [comment]
FYI, here's the list of Web Almanac colours in case you need them for your image:

- Pale green `#a8caba`
- Grey `#62718b`
- Brown `#4a3244`
- Pale yellow - `#ffe599`
- Pale blue - `#9fc5e8`
- Pale orange - `#f6b26b`
- Pale pink - `#ea9999`
- Yellow - `#ffd966`
- Blue - `#6d9eeb`
- Green - `#93c47d`
- Orange - `#e69138`

**Comment by foxdavidj** [comment]
@mel-ada Thanks for the terrific update!

**Comment by foxdavidj** [comment]
@mel-ada Could you and your analysts () hop in the Query Draft PR and add a comprehensive list of all the queries you'd like to run?

If there are a lot of repeats from last year, we can help out by quickly re-running those

**Comment by mel-ada** [comment]
@foxdavidj I've added a list in the README [here](https://github.com/HTTPArchive/almanac.httparchive.org/blob/d6d0ed4ef685739654d7d8f1677b9657d31230a8/sql/2022/performance/README.md#query-list)! I'll defer to @konfirmed and @25prathamesh on if there are similar queries in previous years' chapters.

**Comment by mel-ada** [comment]
@foxdavidj 

🤔 Queries:
Queries were due today. Despite prompts here and on Slack, I haven't heard from @25prathamesh or @konfirmed in a while. I do see some activity on the [queries PR](https://github.com/HTTPArchive/almanac.httparchive.org/pull/2974), but I think it's safe to say this is running late. Analysts, please let us know the latest status and how I can help. **Please note: a delay in queries/data could result in a delay of the rough draft - I'll need the data as soon as possible.** See my above comment for a reminder of my availability.

**Comment by 25prathamesh** [comment]
@mel-ada I have completed the following queries,, my commit got rejected due to linting issue i will fix it 
### Web Getting Wrong
#### Gaming the Metrics (Custom)
- [ ] Filtering for synthetic tests
- [ ] LCP Animation & Overlay Hack
- [ ] LCP Svg Overlay Hack
- [ ] CLS Animation Hack
- [ ] FID iFrame Hack

#### Antipatterns (Custom)
- [ ] LCP Lazy Loaded

 Antipatterns (Not Custom) is pending from my side

**Comment by mel-ada** [comment]
📆 **Timeline Update:**
The rough draft will now be pushed back until 8/22 due to a delay in queries, and I can't commit to including/writing content for data that is missing as of today. This includes the following queries: 
- [ ] % of sites that are using priority as an attribute on LCP element
- [ ] % of sites where the LCP is element is preloaded
- [ ] LCP Element Resource Delay
- [ ] TTFB by Category
- [ ] Domain Sharding
- [ ] FID Double Tap to Zoom Disabled
- [ ] FID / INP Long Task Data (let's pick either FID or INP)

cc reviewers: @konfirmed @rviscomi @25prathamesh @estelle please see my availability above, as I will need to submit the final draft by 9/1.

**Comment by rviscomi** [comment]
Analysis PR is merged and the results are in the [sheet](https://docs.google.com/spreadsheets/d/1TPA_4xRTBB2fQZaBPZHVFvD0ikrR-4sNkfJfUEpjibs/edit?usp=sharing). Over to @mel-ada to start the draft.

**Comment by mel-ada** [comment]
Phew - the way our timeline played out is tight! This has required clear communication on availability and scope negotiation. We are working to deliver the best possible version of the perf chapter within the given constraints, and we've even had some additional help from @siakaramalegos as an analyst and co-author! 🙌 

The draft is still in progress, see our slack channel for the latest. I've also created a placeholder draft markdown PR (linked above). I will wait to add content until final approval, but given my tight timeline of being **unavailable starting 9/1**, and needing to submit the final markdown version by then, I wanted to get branch going. I got a head start on some minor metadata updates. Should full markdown not be in by 9/1, @rviscomi has kindly offered to help out. Fingers crossed this isn't needed, but hard to say since it all depends on when reviewers/editors give final approval.
